digdag-client/src/main/java/io/digdag/client/api/RestSessionAttempt.java

```diff
@@ -48,6 +48,19 @@ default int getIndex()
 
     Optional<Instant> getFinishedAt();
 
+    default String getStatus() {
+        if (this.getCancelRequested()) {
```

Comment:
Ah, it could happen that a task finishes while it received a cancel request right before the finish...? @yoyama 

In this case, I think it should show `success` / `error`.

Reply:
I checked recent 5000000 records of `session_attempts` table in our service

```
 state_flags | count
-------------+--------
           6 | 588183
           2 |  21313
           3 |    635
           0 |    136
```

6 means `success`, 3 means `killed`, 2 means `error` and 0 means `running`. So for now, `success` status that already received a cancel request doesn't exist. But I think it won't be strongly guaranteed on server side design in the future and the change in this PR should be a bit more robust for future changes to some extents. So I'd like to ask you to move `if (this.getCancelRequested()) {...}` into `if (this.getDone()) {...}`.

BTW, I talked with @yoyama about this logic and we thought `cancel requested` and another state should be orthogonal as much as possible. So `success (cancel requested)` or `running (cancel requested)` might need to be added in the future.

Reply:
@komamitsu is this what you meant?

Reply:
I meant it would be better if the condition is like
```
       if (this.getSuccess()) {
            return "success";
        }
        if (this.getDone()) {
            if (this.getCancelRequested()) {
                return "killed";
            }
            else {
                return "error";
            }
        }
        return "running";
```